test/widgets/compose_box_test.dart

```diff
@@ -31,14 +31,27 @@ void main() {
   late PerAccountStore store;
   late FakeApiConnection connection;
 
-  Future<GlobalKey<ComposeBoxController>> prepareComposeBox(WidgetTester tester,
-      {required Narrow narrow, List<User> users = const []}) async {
+  Future<GlobalKey<ComposeBoxController>> prepareComposeBox(WidgetTester tester, {
+    required Narrow narrow,
+    User? selfUser,
+    int? realmWaitingPeriodThreshold,
+    List<User> users = const [],
+    List<ZulipStream> streams = const [],
+  }) async {
+    if (narrow is ChannelNarrow || narrow is TopicNarrow) {
+      final channelId = narrow is ChannelNarrow ? narrow.streamId : (narrow as TopicNarrow).streamId;
```

Comment:
This can be tightened with a fun use of pattern-matching:
```suggestion
    if (narrow case ChannelNarrow(:var streamId) || TopicNarrow(:var streamId)) {
```